src/main/java/ch/njol/skript/events/EvtVehicleCollision.java

```diff
@@ -0,0 +1,109 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.event.Event;
+import org.bukkit.event.vehicle.VehicleBlockCollisionEvent;
+import org.bukkit.event.vehicle.VehicleCollisionEvent;
+import org.bukkit.event.vehicle.VehicleEntityCollisionEvent;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class EvtVehicleCollision extends SkriptEvent {
+
+	static {
+		Skript.registerEvent("Vehicle Collision", EvtVehicleCollision.class, new Class[]{VehicleBlockCollisionEvent.class, VehicleEntityCollisionEvent.class},
+				"vehicle collision [(with|of) [a[n]] %-itemtypes/blockdatas/entitydatas%]",
+			"vehicle block collision [(with|of) [a[n]] %-itemtypes/blockdatas%]",
+			"vehicle entity collision [(with|of) [a[n]] %-entitydatas%]")
+				.description("Called when a vehicle collides with a block or entity.")
+				.examples("on vehicle collision:", "on vehicle collision with obsidian:", "on vehicle collision with a zombie:")
+				.since("INSERT VERSION");
+	}
+
+	private Literal<?> expr;
+	private boolean blockCollision;
+	private boolean entityCollision;
+	private final List<ItemType> itemTypes = new ArrayList<>();
+	private final List<BlockData> blockDatas = new ArrayList<>();
+	private final List<EntityData<?>> entityDatas = new ArrayList<>();
+
+	@Override
+	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult) {
+		if (args[0] != null) {
+			expr = args[0];
+			for (Object object : expr.getArray()) {
```

Comment:
I think you want to use getAll here. For example, if I write `on vehicle collision with dirt or stone`, I probably want it to trigger for either block.